solvers/cd.py

```diff
@@ -66,6 +78,10 @@ def run(self, n_iter):
             L = (self.X ** 2).sum(axis=0)
             self.w = self.cd(self.X, self.y, self.lmbd, L, n_iter)
 
+        if self.fit_intercept:
+            intercept = self.y_offset - self.X_offset @ self.w
```

Comment:
this is going to be wrong for sparse data as the CD needs to do some implicit centering for each coordinate update.

Reply:
Yes, you're right. For now, I'm just adding fit_intercept support for dense data.
The sparse+fit_intercept case is still skipped.

Reply:
Yes, you are right. For now I'm just adding fit_intercept support on the dense case.
The case sparse + fit_intercept is still skipped.

Reply:
it's worth to be explicit and to add `not sparse.issparse(self.X) and self.fit_intercept` to avoid such surprises (all the information is where it's required and the reader does not need to go check `skip`)

Reply:
Ok, will do it !